packages/commonwealth/client/scripts/hooks/api/useTopics.tsx

```diff
@@ -0,0 +1,294 @@
+        queryKeys.TOPICS_FOR_CHAIN(topicsForChainId),
+      ]);
+    },
+    enabled: !!topicsForChainId,
+    staleTime: 60_000, // cache for 60 seconds
+  });
+
+  const addTopicQuery = useMutation({
+    onMutate: async ({
+      name,
+      description,
+      telegram,
+      featuredInSidebar,
+      featuredInNewPost,
+      tokenThreshold,
+      defaultOffchainTemplate,
+    }: {
+      name: string;
+      description: string;
+      telegram: string;
+      featuredInSidebar: boolean;
+      featuredInNewPost: boolean;
+      tokenThreshold: string;
+      defaultOffchainTemplate: string;
+    }) => {
+      // Cancel any outgoing refetches ,(so they don't overwrite
+      // our optimistic update)
+      await queryClient.cancelQueries({
+        queryKey: [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+      });
+
+      // fetch old topics from react query store
+      const oldTopics = queryClient.getQueryData([
+        queryKeys.TOPICS_FOR_CHAIN(topicsForChainId),
+      ]);
+
+      // update optimistic value
+      queryClient.setQueryData(
+        [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+        (t: Topic[]) => {
+          const updatedTopics = [...t];
+
+          const newTopic = new Topic({
+            id: -1, // -1 to indicate a new thread
+            name,
+            description,
+            telegram,
+            featured_in_new_post: featuredInNewPost,
+            featured_in_sidebar: featuredInSidebar,
+            token_threshold: tokenThreshold,
+            default_offchain_template: defaultOffchainTemplate,
+          });
+
+          return [...updatedTopics, newTopic];
+        }
+      );
+
+      return { oldTopics };
+    },
+    mutationFn: async ({
+      name,
+      description,
+      telegram,
+      featuredInSidebar,
+      featuredInNewPost,
+      tokenThreshold,
+      defaultOffchainTemplate,
+    }: {
+      name: string;
+      description: string;
+      telegram: string;
+      featuredInSidebar: boolean;
+      featuredInNewPost: boolean;
+      tokenThreshold: string;
+      defaultOffchainTemplate: string;
+    }) => {
+      const response = await $.post(`${app.serverUrl()}/createTopic`, {
+        chain: app.activeChainId(),
+        name,
+        description,
+        telegram,
+        featured_in_sidebar: featuredInSidebar,
+        featured_in_new_post: featuredInNewPost,
+        default_offchain_template: defaultOffchainTemplate,
+        token_threshold: tokenThreshold,
+        jwt: JWT,
+      });
+
+      const result = new Topic(response.result);
+
+      // update proper value
+      queryClient.setQueryData(
+        [queryKeys.TOPICS_FOR_CHAIN(topicsForChainId)],
+        (t: Topic[]) => {
+          const foundIndex = t.findIndex((x) => x.id === -1);
+
+          t[foundIndex] = new Topic({
+            ...t[foundIndex],
+            ...response.result,
+          });
+
+          return [...t];
+        }
+      );
```

Comment:
when api has a success status, we will replace optimistic data with api response